### DISCUSSION TITLE
Performance issue

### DISCUSSION THREAD

**Discussion by mbnoimi** [pocketbase/pocketbase#1145] [Q&A]
I'm still a newbie so forgive my silly question.

I already read that replacing SQLite isn't considered at all. But when I read [this post](https://medium.com/@kamaljp/pocketbase-and-postgres-19d81f7c6add) I became skeptical of SQLite adopting.

Have you felt in this issue?

Performance issue is a big deal. specially if we compared SQLite to Postgresql.

**Comment by ganigeorgiev** [comment]
It's apples to oranges type of comparison.

The code in the linked post inserts the records one by one synchronously and obviously it will be slower than bulk insertions or sending requests in parallel.

I'll leave a comment later in the linked blog post.

**Reply by mbnoimi** [reply]
Thanks a lot. I'll follow your comment.

**Comment by dschissler** [comment]
As was said, you'd want to do that in batch.  The only real caveat would be if you have file data to process and save with each of those then you'd probably want to do it in an after/post hook.  But you may need to get the data first for validation.  So depending on where the file data came from you'd have other considerations like where you were going to temporary store it in a before/pre hook.  You'd want to prevent file trashing and so it should ideally be temporarily stored on the same filesystem partition so that the actual data isn't moved when you move the file to it's final position.

Anyways, so if you're not having to worry about managing of this processing kind of stuff then you can just do it in a very slow loop like the example if you have all of the time in the world to complete it, or you can do it in a batch operation.  You'd just determine if it's ok if some of the records can fail or if they all need to be perfect.  Or maybe failing somehow is ok and you can just mark a field that it failed.

Also, I've found that with other languages (I don't know how Golang would do) is that anything that needs maximum performance should just completely bypass using an ORM to just use raw SQL to not be creating objects/structs and doing record hooks.  But that is a thing that has very rarely happened to me.

**Reply by dschissler** [reply]
Also, if you need maximum throughput and there are records which cannot fail, but that there is a chance that they might, then you can segment the updates or inserts.  So like 25, 50 or 100 per transaction.  So there are patterns for that.
